include/dsn/dist/failure_detector/failure_detector.h

```diff
@@ -38,6 +38,10 @@
 # include <dsn/dist/failure_detector/fd.client.h>
 # include <dsn/dist/failure_detector/fd.server.h>
 
+namespace dsn { namespace replication {
```

Comment:
This is not appropriate as the low level components should not be aware of the higher layer ones. Better to use other approaches such as mark related methods with protect.